### DISCUSSION TITLE
Consider display classes for touch devices?

### DISCUSSION THREAD

**Discussion by BartzikWebdesign** [twbs/bootstrap#30824] [General]
Please add display classes for touch devices and mousepointer devices like `d-touch-none` and `d-mouse-none` and so on by doing it like this:

`
@media (hover: none) {
   /* for touch devices */ 
  .d-touch-none { ... }
}
@media (hover: hover) {
   /* for mouse devices */ 
  .d-mouse-none { ... }
}
`

This would make it easy to display and hide additional Text on Buttons for example in addition for tooltips which do not work comfortable on mobile devices.

Thanks!

**Comment by mdo** [comment]
Feature requests should be in issues, but for the time being, let's leave this as a discussion. I'll edit the title to reframe things and get some input from others.

**Comment by patrickhlauke** [comment]
i'll note that there are devices that have multiple types of inputs (mouse and touch), and these can even change dynamically (e.g. pairing a mouse with a tablet). for this reason, i'd be against these sorts of classes (unless we also then want to add a whole https://github.com/ten1seven/what-input style detection and dynamic style changes...which would be a bit of a nightmare)
